docs/user-guide.md

```diff
@@ -1,23 +1,26 @@
 # JSpecify user guide
 
-The JSpecify project defines annotations that describe what types in Java code
-can be null. This allows tools to help developers avoid NullPointerExceptions,
-by pinpointing places where those exceptions might happen because the code
-doesn't guard against the possibility that a reference might be null. At the
-same time, the annotations make it possible to declare that certain references
-can never be null.
+In Java code, whether an expression may evaluate to null is often documented
+only in natural language, if at all.  The goal of JSpecify is to permit
+programmers to express specifications (initially, just nullness properties) in a
+machine-readable way.
+
+JSpecify defines annotations that describe whether a Java type contains the value null.
+Such annotations are useful to:
+ * programmers reading the code,
+ * tools that help developers avoid NullPointerExceptions,
+ * tools that perform run-time checking and test generation,
+ * documentation systems,
+ * and for other purposes.
```

Comment:
Thanks, I was unhappy with this initial section and I think this rewrite is much better. My only comment would be a niggle that the last bullet point isn't grammatical ("useful to and for other purposes"). I think it just needs not to be a bullet.

Reply:
Agreed, it bugged me too but I wanted to get the review started.  Thanks for the comment!  I have changed it.